chainerx_cc/chainerx/routines/manipulation.cc

```diff
@@ -781,19 +784,21 @@ Array Moveaxis(const Array& a, const Axes& source, const Axes& destination) {
     std::iota(source_axes.begin(), source_axes.end(), 0);
     std::iota(destination_axes.begin(), destination_axes.end(), 0);
 
-    for (uint8_t i = 0; i < source.ndim(); ++i) {
-        order[destination[i]] = source[i];
-        source_axes[source[i]] = -1;
-        destination_axes[destination[i]] = -1;
+    for (int8_t i = 0; i < source.ndim(); ++i) {
+        order[normalized_destination[i]] = normalized_source[i];
+        source_axes[normalized_source[i]] = -1;
+        destination_axes[normalized_destination[i]] = -1;
     }
 
-    source_axes.erase(std::remove(source_axes.begin(), source_axes.end(), -1), source_axes.end());
-    destination_axes.erase(std::remove(destination_axes.begin(), destination_axes.end(), -1), destination_axes.end());
-    CHAINERX_ASSERT(static_cast<int8_t>(source_axes.size()) == a.ndim() - source.ndim());
-    CHAINERX_ASSERT(static_cast<int8_t>(destination_axes.size()) == a.ndim() - destination.ndim());
+    Axes::iterator source_iter = std::remove(source_axes.begin(), source_axes.end(), -1);
```

Comment:
Google Style encourages to use `auto` for iterators.

Reply:
Sure. Will do that.

Reply:
Done.